tests/jballerina-unit-test/src/test/resources/test-src/expressions/typecast/type-casting.bal

```diff
@@ -27,6 +28,92 @@ function stringtoint(string value) returns int|error {
     return result;
 }
 
+function testByteArrayToIntArray() {
+    byte[] arr = [1, 128, 255];
+    any a = arr;
+    test:assertEquals(a is int[], true);
+    int[] intArray = <int[]> a;
+    test:assertEquals(intArray, [1, 128, 255]);
+}
+
+function testSigned32IntArrayToIntArray() {
+    int:Signed32[] arr = [1, 2, 3];
+    any a =  arr;
+    test:assertEquals(a is int[], true);
```

Comment:
we can use 
```suggestion
    test:assertTrue(a is int[]);
```
Shall we fix in all places?

Reply:
Addressed with the commit https://github.com/ballerina-platform/ballerina-lang/pull/31302/commits/b142a5545959805480887f83416b92b994b128bc